WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListViewModel.swift

```diff
@@ -146,10 +182,36 @@ private extension BlazeCampaignListViewModel {
     /// Updates row view models and sync state.
     func updateResults() {
         campaigns = resultsController.fetchedObjects
+        updateCreateCampaignDestination()
         transitionToResultsUpdatedState()
     }
 }
 
+// MARK: Create campaign navigation
+private extension BlazeCampaignListViewModel {
+    func updateCreateCampaignDestination() {
+        createCampaignDestination = featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation)
+        ? determineDestination()
+        : .webviewForm(productID: productResultsController.fetchedObjects.first?.productID)
+    }
+
+    // Figure out what view should be presented when "Create" button is tapped, depending on the number of eligible products.
+    func determineDestination() -> CreateCampaignDestination {
+        switch productResultsController.fetchedObjects.count {
+        case 0:
+            return .noProductAvailable
+        case 1:
+            guard let product = self.productResultsController.fetchedObjects.first else {
+                // Should not happen since we already checked the count, but just in case.
+                return .noProductAvailable
+            }
+            return .campaignForm(productID: product.productID)
+        default:
+            return .productSelector
+        }
+    }
```

Comment:
What do you think about using if else if? This lets us use `isEmpty` on `fetchedObjects` instead of using hard-coded zero. 


```suggestion
        if productResultsController.fetchedObjects.isEmpty {
            return .noProductAvailable
        } else if productResultsController.fetchedObjects.count == 1,
                  let recentPublishedProduct = productResultsController.fetchedObjects.first {
            return .campaignForm(productID: recentPublishedProduct.productID)
        } else {
            return .productSelector
        }
```

Reply:
I implemented this in the Coordinator https://github.com/woocommerce/woocommerce-ios/pull/11671/files#diff-97e95c754013a21f67871677e78df49649e1cd07195b4f1a5f3128a19ac2516aR95-R104

Thank you!